lib/view_component/compiler.rb

```diff
@@ -92,7 +92,7 @@ def #{method_name}
 
     def define_render_template_for
       variant_elsifs = variants.compact.uniq.map do |variant|
-        "elsif variant.to_sym == :#{variant}\n    #{call_method_name(variant)}"
+        "elsif variant.to_sym == :'#{variant}'\n    send(:'#{call_method_name(variant)}')"
```

Comment:
Can we avoid `send` here somehow (it's slower than "default" method calls)? Maybe we could call `underscore` on `method_name` to make it a valid identifier?

Reply:
@BlakeWilliams Thanks for your comment! I've updated the pull request.